distributed-datawarehouse/duckdb/README.md

```diff
@@ -0,0 +1,253 @@
+
+4. **Query Engine Integration**
+   - Works with any query engine (DuckDB in this case)
+   - Extends single-instance engines to distributed scenarios
+   - Adds orchestration and parallel execution capabilities
+
+## DuckDB Integration
+
+We use DuckDB as our query engine for several reasons:
+
+- Excellent SQL support
+- Direct file reading capabilities
+- Fast analytical processing
+- Simple deployment (single binary)
+
+Bacalhau extends DuckDB's capabilities by:
+
+- Adding distributed orchestration
+- Enabling parallel processing
+- Managing partitioned execution
+- Handling job distribution
+
+### Custom DuckDB Image
+
+Our custom DuckDB image includes User-Defined Functions (UDFs) for partitioning:
+
+- `partition_by_hash`: Even distribution of files
+- `partition_by_regex`: Pattern-based partitioning
+- `partition_by_date`: Time-based partitioning
+
+## Data Access Patterns
+
+Bacalhau supports two ways to access local data:
+
+1. **Direct DuckDB Access**
+
+```sql
+SET VARIABLE files = (
+  SELECT LIST(file)
+  FROM partition_by_hash('s3://local-bucket/**/*.jsonl')
+);
+SELECT * FROM read_json_auto(getvariable('files'));
+```
+
+2. **Bacalhau Input Sources**
+
+```yaml
+InputSources:
+  - Type: s3
+    Source:
+      Bucket: local-bucket
+      Key: "data/*"
+```
+
+Each approach has its benefits:
+
+- Direct access is simpler for SQL-only jobs
+- Input sources provide more control and preprocessing options
+
+## Quick Start
+
+### Prerequisites
+- Bacalhau CLI
+- Docker and Docker Compose
+- Access to example multi-region setup
+
+### Deploy Environment
+```bash
+# Clone and start the multi-region environment
+git clone https://github.com/bacalhau-project/examples.git
+cd examples/setting-up-bacalhau-cluster/docker-compose/multi-region
+docker compose up -d
+```
+
+### Generate Sample Data
+```bash
+# Generate US region data
+bacalhau job run -V Region=us -V Events=3000 \
+  -V StartDate=2024-01-01 -V EndDate=2024-12-31 \
+  -V RotateInterval=month data-generator.yaml
+
+# Generate EU region data
+bacalhau job run -V Region=eu -V Events=3000 \
+  -V StartDate=2024-01-01 -V EndDate=2024-12-31 \
+  -V RotateInterval=month data-generator.yaml
+```
+
+### Run Your First Analysis
+```bash
+# Run monthly trends analysis
+bacalhau job run -V Region=us monthly-trends.yaml
+```
+
+Sample output:
+```
+month       | total_txns | revenue  | unique_customers | avg_txn_value
+------------|------------|----------|------------------|---------------
+2024-03-01  |      3,421 | 178,932  |           1,245 |         52.30
+2024-02-01  |      3,156 | 165,789  |           1,189 |         52.53
+2024-01-01  |      2,987 | 152,456  |           1,023 |         51.04
+```
+   
+
+## Analysis Jobs
+
+### Operational Monitoring
+
+1. **Hourly Operations** (`hourly-operations.yaml`)
+
+- Tracks operational health metrics
+- Monitors transaction success rates
+- Shows hourly patterns
+
+```bash
+bacalhau job run -V Region=us hourly-operations.yaml
```

Comment:
Lol, what is "-V" ? I've never seen it before

Reply:
It sets template variable :)
https://docs.bacalhau.org/references/jobs/job-templating